packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -1082,6 +1088,32 @@ class RawGestureDetectorState extends State<RawGestureDetector> {
     _syncAll(widget.gestures);
   }
 
+  @override
+  void didChangeAccessibilityFeatures() {
+    if (!widget.excludeFromSemantics)
+      _maybeUpdateSemanticsActionOverrides();
+  }
+
+  void _maybeUpdateSemanticsActionOverrides() {
+    final RenderSemanticsGestureHandler semanticsGestureHandler = _getSemanticsGestureHandler();
+    final Set<CustomSemanticsAction> overrides = <CustomSemanticsAction>{};
+    if (WidgetsBinding.instance.window.accessibilityFeatures.switchControl && semanticsGestureHandler.validActions != null) {
```

Comment:
Why do we need to do this conditionally on whether switchControll is on or off? Just providing a label for the actions should be fine either way, no?

Reply:
This is not just overriding the semantics action label, it actually creates a custom action with that label and delegate that custom action to the call the semantics action it overrides.  We can of course do the check on the engine side, but i feel it should the the framework that decides whether we want to add that or not.